matborowiak-tutorial/part five (#13)
gatsbyjs/gatsby-pl: docs/tutorial/part-five/index.md

disableTableOfContents: true
---

> > Ten poradnik jest częścią serii na temat warstw danych Gatsby. Upewnij się, że zapoznałeś się z [częścią czwartą](/tutorial/part-four/) znaim będziesz kontynuował tutaj.

Reviewer: ```suggestion
> Ten poradnik jest częścią serii na temat warstwy danych w Gatsby. Upewnij się, że zapoznałeś się z [częścią czwartą](/tutorial/part-four/) znaim będziesz kontynuował.
```
Ja bym chyba tak dał, bo to jest jedna warstwa. I te 'tutaj' na końcy chyba też jakoś dziwnie brzmi.

> > Ten poradnik jest częścią serii na temat warstw danych Gatsby. Upewnij się, że zapoznałeś się z [częścią czwartą](/tutorial/part-four/) znaim będziesz kontynuował tutaj.

Author: ```suggestion
> Ten poradnik jest częścią serii na temat warstw danych Gatsby. Upewnij się, że zapoznałeś się z [częścią czwartą](/tutorial/part-four/) zanim będziesz kontynuował tutaj.
```

## Czego nauczysz się w tym poradniku?

> W tym poradniku dowiesz się, jak pobierać dane do strony Gatsby przy użyciu GraphQL i wtyczek źródłowych. Zanim jednak dowiesz się o tych wtyczkach, najpierw musisz wiedzieć, jak korzystać z czegoś co nazywa się GraphiQL - narzędzia, które pomaga ułorzyć strukturę Twoich zapytań.

Reviewer: ```suggestion
W tym poradniku dowiesz się, jak pobierać dane do strony Gatsby przy użyciu GraphQL i wtyczek źródłowych. Zanim jednak dowiesz się o tych wtyczkach, najpierw musisz wiedzieć, jak korzystać z czegoś co nazywa się GraphiQL - narzędzia, które pomaga ułożyć strukturę Twoich zapytań.
```

## Przedstawiamy GraphiQL

GraphiQL jest zintegrowanym środowiskiem deweloperskim (IDE) w GraphQL. Jest to potężne (i pod wieloma względami niesamowite) narzędzie, którego będziesz często używać podczas tworzenia stron internetowych w Gatsby.

Masz do niego dostęp, gdy Serwer Deweloperski Twojej strony działa zwyczajnie pod adresem
<http://localhost:8000/___graphql>.

<video controls="controls" autoplay="true" loop="true">
  <source type="video/mp4" src="/graphiql-explore.mp4"></source>
  <p>Twoja przeglądarka nie obsługuje tego elementu wideo.</p>
</video>

> Rozejrzyj się po "typie" `Site` i sprawdź jakie pola są dostępne -- sprawdź też obiekt `siteMetadata` który którkego zapytanie wykonałeś wcześniej. Spróbuj otworzyć GraphiQL i pobawić się trochę ze swoimi danymi! Wciślnij <kbd>Ctrl + Spacja</kbd> (lub <kbd>Shift + Spacja</kbd> jako alternatywny skrót), żeby wyświetlić okno autouzupełniania i <kbd>Ctrl + Enter</kbd>, aby wykonać zapytanie GraphQL. Będziesz używał GraphQL znacznie więcej w dalszej części poradnika.

Reviewer: ```suggestion
Zapoznaj się z "typem" `Site` i sprawdź jakie pola są dostępne -- sprawdź też obiekt `siteMetadata` do którego zapytanie wykonałeś wcześniej. Spróbuj otworzyć GraphiQL i pobawić się trochę ze swoimi danymi! Wciśnij <kbd>Ctrl + Spacja</kbd> (lub <kbd>Shift + Spacja</kbd> jako alternatywny skrót), żeby wyświetlić okno autouzupełniania i <kbd>Ctrl + Enter</kbd>, aby wykonać zapytanie GraphQL. Będziesz używał GraphiQL znacznie częściej w dalszej części poradnika.
```

## Używanie Eksploratora GraphiQL

> Eksplorator GraphiQL pozwoli Tobie interaktywnie budować kompletne zapytania poprzez kliknięcia w dostępne pola oraz wejścia/inputy, oszczędzając Ci żmudnego procesu wpisywania tych zapytań ręcznie na klawiaturze.

Reviewer: ```suggestion
Eksplorator GraphiQL pozwoli Ci interaktywnie budować kompletne zapytania poprzez kliknięcia w dostępne pola oraz wejścia/inputy, oszczędzając Ci żmudnego procesu wpisywania tych zapytań ręcznie na klawiaturze.
```

<EggheadEmbed
  lessonLink="https://egghead.io/lessons/gatsby-build-a-graphql-query-using-gatsby-s-graphiql-explorer"
  lessonTitle="Build a GraphQL Query using Gatsby’s GraphiQL Explorer"
/>

> ## Wtyczki Sources

Reviewer: ```suggestion
## Wtyczki źródłowe
```

Author: ```suggestion
## Wtyczki Źródeł
```
Tak chyba lepiej w kontekście tego czym są te wtyczki.

Wydaje mi się, że Source plugins, mogą być nazwą własną wprowadzoną przez Gatsby... Mam opór przed tłumaczeniem wszystkiego dosłownie, bo czasami gdy sam czytam polskie słowotwórstwo to nie rozumiem połowy tekstu (nie umiem zwizualizować sobie o czym ten tekst mówi, bo nie napotykam się praktycznie na te słowa, w takim kontekście, w codziennej mowie i wtedy przełączam na wersję angielską. (`wtyczka źródła` brzmi jak jakaś końcówka od węża ogrodowego, a przecież chyba sami wszyscy mówimy po polsku zapożyczone z ang `pluginy`)
W dalszym tłumaczeniu starałem się budować tak zdania żeby używać słowa "źródło" w kontekście Source plugins i osiągnąć pewien kompromis, jednak sam termin uważam że brzmi lepiej w oryginale i możemy go potraktować jako nazwę własną pewnej rodziny pluginów Gatsby.
Co o tym myślicie? @lekterable @maciekgrzybek

Reviewer: Wtyczki Źródeł brzmi lepiej niż Wtyczki Źródłowe, to fakt :) Byłbym za taka formą.

> Dane w stronach Gatsby mogą pochodzić zewsząd: API, baz danych, CMS, plików lokalnych, itp.

Reviewer: ```suggestion
Dane w stronach Gatsby mogą pochodzić z różnych źródeł: API, bazy danych, CMS, pliki lokalne, itp.
```

> Wtyczki Sources pobierają dane ze swoejgo źródła. Np. wtyczka `filesystem source` wie jak pobierać dane z systemu plików. Wtyczka WordPress wie jak pobierać dane z WordPress API.

Reviewer: ```suggestion
Wtyczki źródłowe pobierają dane ze swojego źródła. Np. wtyczka `filesystem source` wie jak pobierać dane z systemu plików. Wtyczka WordPress wie jak pobierać dane z WordPress API.
```

Author: ```suggestion
Wtyczki Źródeł pobierają dane ze swojego źródła. Np. wtyczka `filesystem source` wie jak pobierać dane z systemu plików. Wtyczka WordPress wie jak pobierać dane z WordPress API.
```

> Dodaj wtyczkę [`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/) i sprawdź jak ona działa.

Reviewer: ```suggestion
Dodaj wtyczkę [`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/) i sprawdź jak działa.
```

}
```

Zapisz plik i uruchom serwer deweloperski Gatsby. Następnie otwórz ponownie GraphiQL.

W panelu Explorer, zobaczysz `allFile` oraz `file` jako dostępny wybór:

![graphiql-filesystem](graphiql-filesystem.png)

> Kliknij w menu `allFile`. Ustaw kursor zaraz za `allFile` w obszarze zapytania, a następnie wciśnij <kbd>Ctrl + Enter</kbd>. To sprawi autouzupełnienie zapytania o `id` każdego z plików. Wciśnij "Play" aby wykonać zapytanie:

Reviewer: ```suggestion
Kliknij w menu `allFile`. Ustaw kursor zaraz za `allFile` w obszarze zapytania, a następnie wciśnij <kbd>Ctrl + Enter</kbd>. Uruchomi to autouzupełnienie zapytania o `id` każdego z plików. Wciśnij "Play" aby wykonać zapytanie:
```

> Kliknij w menu `allFile`. Ustaw kursor zaraz za `allFile` w obszarze zapytania, a następnie wciśnij <kbd>Ctrl + Enter</kbd>. To sprawi autouzupełnienie zapytania o `id` każdego z plików. Wciśnij "Play" aby wykonać zapytanie:

Author: ```suggestion
Kliknij w menu `allFile`. Ustaw kursor zaraz za `allFile` w obszarze zapytania, a następnie wciśnij <kbd>Ctrl + Enter</kbd>. To sprawi autouzupełnienie zapytania o `id` każdego z plików. Naciśnij "Play" aby wykonać zapytanie:
```

![filesystem-query](filesystem-query.png)

> W panelu Explore, pole `id` zostało automatycznie zaznaczone. Wybierz więcej kategorii pól klikając w odpowiadające im pola wyboru. Wciśnij "Play" aby wykonać zapytanie, ponownie, tym razem z nowymi polami:

Author: ```suggestion
W panelu Explore, pole `id` zostało automatycznie zaznaczone. Wybierz więcej kategorii pól klikając w odpowiadające im pola wyboru. Naciśnij "Play" aby wykonać zapytanie, ponownie, tym razem z nowymi polami:
```

![filesystem-explorer-options](filesystem-explorer-options.png)

> Alternatywnie, możesz dodawać pola używając skrótu autouzupełnienia (<kbd>Ctrl + Space</kbd>). To pokaże wszystkie dostępne pola dla zapytań na Węzłach (nodes) `File`.

Reviewer: Te węzły jakos nie pasują, ale szczerze mówiąc, nie mam za bardzo żadnej alternatywy. @lekterable @matborowiak  jakiś pomysł?

Author: j.w. uważam że pewnych terminów nie da się tłumaczyć dosłownie, i powinniśmy korzystać z "zapożyczeń" z angielskiego. Nie popadajmy w paranoje, skoro sami na codzien w języku polskim używamy słów jak "pluginy" "string" "array" "nody" itp. Jak będziemy tłumaczyć wszystko dosłownie to będzie to wyglądało nie dość że komicznie to jeszcze będzie nie do zrozumienia, bo te polskie słowa praktycznie nigdy w mowie codziennej nie występują w takim kontekście, a precież chodzi o to żeby dokumentacja była zrozumiała dla nas, Polaków XXI wieku. Takie moje zdanie. Co sądzicie?

Reviewer: Zgadzam się jak najbardziej. Node (nie jakos node.js tylko ogólnie) zostawiłbym jako node.

Author: > Zgadzam się jak najbardziej. Node (nie jakos node.js tylko ogólnie) zostawiłbym jako node.

Ok, dodam też tego typu słówka z części piątej poradnika do słowniczka na swoim branchu. Jak zrobimy merge ostatecznej wersji części piątej, to zrobię PR z nowymi terminami. Proponuje żeby wszyscy tak robili, bo często jedna część poradnika nawiązuje do drugiej, i warto żeby terminologia się pokrywała, dlatego jeśli część nad którą pracujesz wprowadza jakieś nowe słówka, dodawaj je na swoim branchu do słowniczka, aby kazda nastepna osoba mogla z nich momentalnie korzystać

Author: ```suggestion
Alternatywnie, możesz dodawać pola używając skrótu autouzupełnienia (<kbd>Ctrl + Space</kbd>). To pokaże wszystkie dostępne pola dla zapytań do `File` Node.
```

![filesystem-autocomplete](filesystem-autocomplete.png)

Spróbuj teraz dodać większą liczbę pól do twojego zapytania, wciskając po każdym razie <kbd>Ctrl + Enter</kbd>

> by wykonywać kolejne zapytania. Zobaczysz akutalizujące się wyniki zapytań:

Reviewer: ```suggestion
by wykonywać kolejne zapytania. Zobaczysz aktualizujące się wyniki zapytań:
```

![allfile-query](allfile-query.png)

Wynikiem jest tablica "węzłów" `File` (węzeł to takie ekstrawaganckie słówko na obiekt w "graph").
Każdy obiekt-węzeł `File` zawiera pola dla których wykonałeś zapytanie.

> ## Zbuduj stronę za pomocą zapytania GraphQL

Reviewer: ```suggestion
## Zbuduj stronę używając zapytań GraphQL
```

> Budowanie nowych stron z Gatsby często rozpoczyna się w GraphiQL. Zaczynasz najpierw

Reviewer: ```suggestion
Budowanie nowych stron z Gatsby często rozpoczyna się w GraphiQL. Zaczynasz od utworzenia
szkic zapytań w GraphiQL (metodą prób i błędów),
a potem kopiujesz działające już zapytania do komponentu strony React by rozpocząć budowę interfejsu.
```
Może tak?

Author: ```suggestion
Budowanie nowych stron z Gatsby często rozpoczyna się w GraphiQL. Zaczynasz od utworzenia
szkicu zapytań w GraphiQL (metodą prób i błędów),
a potem kopiujesz działające już zapytania do komponentu strony React by rozpocząć budowę UI.
```

tworzyć szkic zapytań o dane metodą prób i błędów w GraphiQL,
a potem kopiujesz działające już zapytania do komponentu strony React by rozpocząć budowę UI.

> Spróbujmy tego.

Author: ```suggestion
Spróbujmy tak zrobić.
```

`
```

Wyróżniliśmy powyżej linię `console.log(data)`. Gdy tworzymy nowy komponent,
często pomocnym okazuje się, by wyświetlić w konsoli dane które otrzymujemy z zapytań do GraphQL,
aby można było sprawdzać sobie te dane w konsoli przeglądarki, podczas budowania UI.

Jeśli odwiedzisz nową stronę pod adresem `/my-files/` i otworzysz konsolę przeglądarki której używasz,

> zobaczysz coś co wygląda mniej więcej tak tak:

Reviewer: ```suggestion
zobaczysz coś co wygląda mniej więcej tak:
```

![data-in-console](data-in-console.png)

> Forma otrzymanych danych odpowiada formie zapytania GraphQL.

Reviewer: ```suggestion
Format otrzymanych danych odpowiada formatowi zapytania GraphQL.
```

> Dodaj teraz kod do twojego komponentu by wyświetlić dane Pliku.

Author: ```suggestion
Dodaj teraz kod do twojego komponentu aby wyświetlić dane Pliku.
```

`
```

A teraz wejdź pod adres [http://localhost:8000/my-files](http://localhost:8000/my-files)… 😲

![my-files-page](my-files-page.png)

> ## Czego nauczysz się w nastęnej części?

Author: ```suggestion
## Czego nauczysz się w następnej części?
```

> Teraz wiesz już jak wtyczki Source wprowadzają dane _do wnętrza_ sytemu danych Gatsby. W kolejnym poradniku nauczymy cię jak wtyczki Transformer _transformują_ surowe dane wprowadzone przez wtyczki Source. Kombinacja wtyczek Source i Transformer potrafi kontrolować wszystkie źródła oraz transformacje danych, których możesz potrzebować budując strony Gatsby. Naucz się wtyczek Transformer w [części szóstej poradnika](/tutorial/part-six/).

Reviewer: ```suggestion
Teraz wiesz już jak, wtyczki źródłowe wprowadzają dane _do wnętrza_ sytemu danych Gatsby. W kolejnym poradniku nauczymy cię jak wtyczki transformujące _transformują_ surowe dane wprowadzone przez wtyczki źródłowe. Kombinacja wtyczek źródłowych i transformujących potrafi kontrolować wszystkie źródła oraz transformacje danych, których możesz potrzebować budując strony Gatsby. Naucz się wtyczek transformujących w [części szóstej poradnika](/tutorial/part-six/).
```

Author: ```suggestion
Teraz wiesz już jak wtyczki Źródeł wprowadzają dane _do wnętrza_ sytemu danych Gatsby. W kolejnym poradniku nauczymy cię jak wtyczki Transformacji _transformują_ surowe dane wprowadzone przez wtyczki Źródeł. Kombinacja wtyczek Źródeł i Transformacji potrafi kontrolować wszystkie źródła oraz transformacje danych, których możesz potrzebować budując strony Gatsby. Naucz się wtyczek Transformacji w [części szóstej poradnika](/tutorial/part-six/).
```

> Teraz wiesz już jak wtyczki Source wprowadzają dane _do wnętrza_ sytemu danych Gatsby. W kolejnym poradniku nauczymy cię jak wtyczki Transformer _transformują_ surowe dane wprowadzone przez wtyczki Source. Kombinacja wtyczek Source i Transformer potrafi kontrolować wszystkie źródła oraz transformacje danych, których możesz potrzebować budując strony Gatsby. Naucz się wtyczek Transformer w [części szóstej poradnika](/tutorial/part-six/).

Author: ```suggestion
Teraz wiesz już jak wtyczki Źródeł wprowadzają dane _do wnętrza_ systemu danych Gatsby. W kolejnym poradniku nauczymy Cię jak wtyczki Transformacji _transformują_ surowe dane wprowadzone przez wtyczki Źródeł. Kombinacja wtyczek Źródeł i Transformacji potrafi kontrolować wszystkie źródła oraz transformacje danych, których możesz potrzebować budując strony Gatsby. Naucz się wtyczek Transformacji w [części szóstej poradnika](/tutorial/part-six/).
```
